### PR TITLE
Xnat API adaptions

### PR DESCRIPTION
**Pull request by AndreasFetzer**
Short summary of the changes:

* Added new file ctkXnatConstanst that contains values of the different xnatObject fields
* Replaced calls which access properties to use the values defined in ctkXnatConstants
* xnatListModel and xnatTreeModel displayed the data slightly different. Now it is consistently done
* Fields that must not be synchronized are now stored within the private implementations (e.g. description, which is used as tooltip)
* Inheritance from the xnatListModel was not possible since the rootObject was private. Now it is protected.
* Some refactoring (e.g. alphabetic ordering of the includes, …)

### CODE REVIEW

File: Libs/XNAT/Core/ctkXnatFile.cpp [hunk 1]
```diff
@@ -56,9 +57,54 @@ ctkXnatFile::~ctkXnatFile()
 }
 
 //----------------------------------------------------------------------------
+void ctkXnatFile::setName(const QString &name)
+{
+  setProperty(ctkXnatObjectFields::FILE_NAME, name);
```
Comment (jcfr) [inline Libs/XNAT/Core/ctkXnatFile.cpp]: `this->setProperty(... `

File: Libs/XNAT/Core/ctkXnatFile.cpp [hunk 2]
```diff
@@ -56,9 +57,54 @@ ctkXnatFile::~ctkXnatFile()
...
+}
+
+//----------------------------------------------------------------------------
+QString ctkXnatFile::name() const
+{
+  return property(ctkXnatObjectFields::FILE_NAME);
```
Comment (jcfr) [inline Libs/XNAT/Core/ctkXnatFile.cpp]: `return this->property(...`

**Comment by jcfr** [comment]
Looks good to me (beside of few stylistic comments)

Some other XNat user should probably review. @ntoussaint @nolden  What do you think ?

**Comment by espakm** [comment]
id() and setId() should be virtual, so that ctkXnatResource can overwrite it them. I left comments on the lines.

I agree with replacing the string literals with named constants. However, I would define the constants in ctkXnatObject and its derived classes, wherever it is appropriate. What do you think? Does any class need access to those constants other than ctkXnatObject and its derivatives? Or will it be required later?

Other than that it looks good.

**Comment by AndreasFetzer** [comment]
Regarding the constants:
I think if you deal with an xnatObject and want to access certain properties then you might not want to include all the derived classes in order to get the respective property identifiers.

A suggestion would be to create separate structs in the file ctkXnatConstants so that not all the fields are accessed via ctkXnatObejctFields::SOMETHING but e.g. ctkXnatFileFields::FORMAT, ctkXnatScanFields::QUALITY, …

Anyway if you think the better solution would be to move the constants into the respective class implementations then I can do that.

**Comment by espakm** [comment]
> I think if you deal with an xnatObject and want to access certain properties
> then you might not want to include all the derived classes in order to get
> the respective property identifiers.

That was my question about. Do you need to refer to field names from a class that is not derived from ctkXnatObject?

Also, if you want to use a field of e.g. ctkXnatFile, it is likely that you want to store a ctkXnatFile object, so ctkXnatFile.h would be included anyway. And so would ctkXnatObject.h, transitively.

**Comment by AndreasFetzer** [comment]
Yes you are right! I think having the properties located in the respective implementation is more reasonable. I made the changes accordingly.

**Comment by espakm** [comment]
Great, thanks!
No other thoughts, it can be merged. I have no rights.

**Comment by espakm** [comment]
Great, thanks!
No other thoughts, it can be merged. I have no rights.

**Comment by espakm** [comment]
Great, thanks!
No other thoughts, it can be merged. I have no rights.

**Comment by jcfr** [comment]
:+1:  Great team discussion :)

**Pull request merged**